cache-tree.c

```diff
@@ -513,7 +526,12 @@ static void write_one(struct strbuf *buffer, struct cache_tree *it,
 
 void cache_tree_write(struct strbuf *sb, struct cache_tree *root)
 {
+	trace2_region_enter("cache_tree", "write_one", NULL);
+
 	write_one(sb, root, "", 0);
+
+	trace2_data_intmax("cache_tree", NULL, "write_size", (intmax_t)sb->len);
```

Comment:
It probably does not matter on the platforms we care about (i.e. 64-bit ones), but technically a very large `sb->len` could overflow to a negative value in the trace.

It might make more sense to introduce a `trace2_data_uintmax()` and use that. (But I won't insist on that.)